test/e2e/specs/editor/plugins/post-type-locking.spec.js

```diff
@@ -0,0 +1,455 @@
+			await editor.canvas
+				.getByRole( 'button', {
+					name: 'Two columns; equal split',
+				} )
+				.click();
+
+			await expect(
+				page
+					.getByRole( 'document', {
+						name: 'Block: Column (1 of 2)',
+					} )
+					.getByRole( 'button', { name: 'Add block' } )
+			).toBeHidden();
+
+			await expect(
+				page
+					.getByRole( 'toolbar', { name: 'Document tools' } )
+					.getByRole( 'button', { name: 'Toggle block inserter' } )
+			).toBeDisabled();
+		} );
+	} );
+
+	test.describe( 'template_lock insert', () => {
+		test.beforeEach( async ( { admin } ) => {
+			await admin.createNewPost( { postType: 'locked-insert-post' } );
+		} );
+
+		test( 'should disable the inserter', async ( { page } ) => {
+			await expect(
+				page
+					.getByRole( 'toolbar', { name: 'Document tools' } )
+					.getByRole( 'button', { name: 'Toggle block inserter' } )
+			).toBeDisabled();
+		} );
+
+		test( 'should not allow blocks to be removed', async ( {
+			editor,
+			page,
+		} ) => {
+			await editor.canvas
+				.getByRole( 'document', {
+					name: 'Empty block',
+				} )
+				.first()
+				.fill( 'p1' );
+
+			await editor.clickBlockToolbarButton( 'Options' );
+
+			await expect(
+				page
+					.getByRole( 'menu', { name: 'Options' } )
+					.getByRole( 'menuitem', { name: 'Delete' } )
+			).toBeHidden();
+		} );
+
+		test( 'should allow blocks to be moved', async ( { editor, page } ) => {
+			await editor.canvas
+				.getByRole( 'document', {
+					name: 'Empty block',
+				} )
+				.first()
+				.click();
+
+			await page
+				.getByRole( 'toolbar', { name: 'Block tools' } )
+				.getByRole( 'button', { name: 'Move up' } )
+				.click();
+
+			await expect.poll( editor.getBlocks ).toMatchObject( [
+				{
+					name: 'core/paragraph',
+				},
+				{
+					name: 'core/image',
+				},
+				{
+					name: 'core/quote',
+				},
+				{
+					name: 'core/columns',
+				},
+			] );
+		} );
+	} );
+
+	test.describe( 'template_lock false', () => {
+		test.beforeEach( async ( { admin } ) => {
+			await admin.createNewPost( { postType: 'not-locked-post' } );
+		} );
+
+		test( 'should allow blocks to be inserted', async ( {
+			editor,
+			page,
+		} ) => {
+			await expect(
+				page
+					.getByRole( 'toolbar', { name: 'Document tools' } )
+					.getByRole( 'button', { name: 'Toggle block inserter' } )
+			).toBeEnabled();
+
+			await editor.insertBlock( { name: 'core/list' } );
+			await editor.canvas
+				.getByRole( 'textbox', { name: 'List text' } )
+				.fill( 'List content' );
+
+			await expect(
+				editor.canvas.getByRole( 'document', {
+					name: 'Block: List item',
+				} )
+			).toHaveText( 'List content' );
+		} );
+
+		test( 'should allow blocks to be removed', async ( { editor } ) => {
+			await editor.canvas
+				.getByRole( 'document', {
+					name: 'Empty block',
+				} )
+				.first()
+				.fill( 'p1' );
+
+			await editor.clickBlockOptionsMenuItem( 'Delete' );
+			await expect.poll( editor.getBlocks ).not.toMatchObject( [
```

Comment:
Could we instead show what it matches?

Reply:
Just to ensure the action doesn't wipe out the whole content?

Reply:
Generally speaking checking things "positively" makes sure we are testing the right thing because there are too many ways a result could "not" be something 😆. Also maybe a little easier to read. Using "not" is totally viable though in some cases where we're comparing the result with an already initialized variable.

Reply:
Makes sense. I'll update the assertion.

Reply:
Updated in d7f6917.